editor.planx.uk/src/pages/FlowEditor/components/Settings/Submissions/index.tsx

```diff
@@ -1,9 +1,35 @@
 import Box from "@mui/material/Box";
 import Typography from "@mui/material/Typography";
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { FeaturePlaceholder } from "ui/editor/FeaturePlaceholder";
 
+import { useStore } from "../../../lib/store";
+import {
+  fetchSubmittedApplications,
+  SubmissionData,
+} from "./submissionDataTypesAndQueries";
+
 const Submissions: React.FC = () => {
+  const flowSlug = useStore.getState().flowSlug;
+  const teamSlug = useStore.getState().teamSlug;
+  const [applications, setApplications] = useState<SubmissionData[]>();
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    if (flowSlug && teamSlug) {
+      fetchSubmittedApplications(flowSlug, teamSlug)
+        .then((result) => setApplications(result.submission_services_summary))
+        .catch((error) => {
+          setError(error);
+        });
+    }
+  }, [flowSlug, teamSlug]);
```

Comment:
Data fetching from within components is probably not handled very consistently currently in the application. I would think most data queries are handled at the route level, set to the store, and then updated from there.

In this instance, I'd recommend you take a look at the `useQuery()` hook which allows access to nice loading and error variables which can be used by the UI - this might allow you to simplify some of this code.